### PR TITLE
Release azure cosmos 4.9.0

### PR DESCRIPTION
**Pull request by kushagraThapar**
azure-cosmos ->
#### New Features
* Added Beta API for Bulk Operations.
* Added `getRegionsContacted` API in `CosmosDiagnostics`.
* Added Diagnostics for `CosmosStoredProcedureResponse`.
* Added trouble shooting guide links to `CosmosException`.

#### Key Bug Fixes
* Adding automatic retries on client-side transient failures on writes while possible with still being idempotent.
* Fixed NPE on `getDiagnostics` for `CosmosStoredProcedureResponse`.
* Fixed empty `resourceAddress` in `CosmosException`.

azure-spring-data-cosmos ->
#### New Features
* Updated Spring Data Commons version to 2.3.5.RELEASE.
* Updated Spring Core version to 5.2.10.RELEASE.

#### Key Bug Fixes
* Fixed publishing of `spring.factories` file with released jar.
* Fixed repository query with repeated parameters.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,21 @@
 ## Release History
 
-## 4.9.0-beta.1 (Unreleased)
+### 4.9.0 (2020-12-11)
+#### New Features
+* Added Beta API for Bulk Operations.
+* Added `getRegionsContacted` API in `CosmosDiagnostics`.
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:6]: `Set<URI> CosmosDiagnostics.getRegionsContacted()`
I might have missed this earlier in the PR.

two comments:
1. For `CosmosClientBuilder.endpoint(String)` we pass the endpoint as String not URI, but in the above newly added API we return the endpoint as URI. seems inconsistent to me.
2. For other apis when we return a collection we return Iterable not Set.

Can we tag this new API with Beta annotation for now. So if we can have the option of changing it later?

**Review by moderakh — APPROVED** [review]
I have one comment except that LGTM.

the new API: `Set<URI> CosmosDiagnostics.getRegionsContacted()`
seems I might have missed this earlier in the PR.

two comments:
1. For `CosmosClientBuilder.endpoint(String)` we pass the endpoint as String not URI, but in the above newly added API we return the endpoint as URI. seems inconsistent to me.
2. For other apis when we return a collection we return Iterable not Set.

Can we tag this new API with Beta annotation for now. So if we can have the option of changing it later?

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:6]: Makes sense, will add the beta annotation to it.

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:6]: Done.

**Pull request merged**